src/script/Monitor/GenerateMONIT.py

```diff
@@ -265,16 +266,40 @@ def execute(self):
 
     # before running make sure no other instance of this script is running
     lockFile = workdir + 'CRAB3_SCHEDD_JSON.Lock'
+    currentPid = str(os.getpid())
+
+    # Check if lockfile already exists and if it does, check PID number in the lockfile if it is running
     if os.path.isfile(lockFile):
-        logger.error("%s already exists, abandon this run." % lockFile)
-        print("%s already exists, abandon this run" % lockFile)
-        exit()
-    else:
-        open(lockFile, 'wa').close()  # create the lock
-        logger.info('Lock created. Start data collection')
+        if os.stat(lockFile).st_size == 0:
+            logger.error("Lockfile is there but it is empty. Removing old lockfile.")
+            os.remove(lockFile)
+        else:
+            with open(lockFile, 'r') as lf:
+                pid = int(lf.read())
+            try:
+                os.kill(pid, 0)
+            except OSError as e:
+                if e.errno == errno.ESRCH:  # ESRCH - No such process
+                    logger.error("Lockfile is there but program is not running. Removing old lockfile.")
+                    os.remove(lockFile)
+                elif e.errno == errno.EPERM:  # EPERM - Operation not permitted (i.e., process exists)
+                    logger.error("Operation not permitted (i.e., process exists), abandon this run.")
```

Comment:
You may put a comment where you note that EPERM is raised when process does not exist, but in the log I would only write "Previous process still running". If you are not sure that this check works.. can always fork a  `ps -p pid` and check it stdout.

Reply:
Updated log comment